src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtil.java

```diff
@@ -97,7 +97,7 @@ public static Map<String, Integer> nameToValueMapFromPublicIntFields(Class<?> cl
         return Arrays.stream(cls.getDeclaredFields())
             .filter(fld -> Modifier.isPublic(fld.getModifiers()) && fld.getType() == Integer.TYPE)
             .collect(Collectors.toUnmodifiableMap(
-                Field::getName, fld -> getIntFromField(fld, fld.getName()))
+                Field::getName, fld -> getIntFromField(fld, fld))
```

Comment:
`getIntFromField` is basically a wrapper for `Field#getInt`. From reading the docs of [Field](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/Field.html#getInt(java.lang.Object)), it looks like we should be passing the class itself to this method. I am not sure how this ever worked in the first place. Please do:

```suggestion
                Field::getName, fld -> getIntFromField(fld, cls))
```

Reply:
> I am not sure how this ever worked in the first place

This is because we are going against a static field. It has no instance to speak of, so most likely reflection is ignoring the 2nd parameter we pass. We also have similar funky uses in `TokenUtilTest` as `getIntFromField` is `public` when it should not be.

Truthfully this 2nd parameter should be `null`, as there is no instance.

```suggestion
                Field::getName, fld -> getIntFromField(fld, null))
```

Reply:
@rnveach leaving this for you to resolve.